### PR TITLE
Support global `repositories.azure.` settings

### LINKED ISSUE #13776: `repositories.azure.base_path` is not used
When looking at azure code, `repositories.azure.base_path` is actually never read although we said it is in documentation.

See also https://github.com/elastic/elasticsearch-cloud-azure/issues/98

### PR DESCRIPTION
**Pull request by dadoonet**
All those repository settings can also be defined globally in `elasticsearch.yml` file using prefix `repositories.azure.`. For example:

``` yml
repositories.azure:
    container: backup-container
    base_path: backups
    chunk_size: 32m
    compress": true
```

Closes #13776.

### CODE REVIEW

**Comment by dadoonet** [comment]
@imotov or @tlrx Wanna review it please?

File: plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java [hunk 1]
```diff
@@ -37,9 +37,12 @@
     final class Storage {
         public static final String PREFIX = "cloud.azure.storage.";
         @Deprecated
-        public static final String ACCOUNT = "cloud.azure.storage.account";
+        public static final String ACCOUNT_DEPRECATED = "cloud.azure.storage.account";
         @Deprecated
-        public static final String KEY = "cloud.azure.storage.key";
+        public static final String KEY_DEPRECATED = "cloud.azure.storage.key";
```
Comment (imotov) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: Should we be filtering out this setting when we display settings over REST API?

**Comment by imotov** [comment]
Left one comment about filtering out the key setting. Otherwise, LGTM.

File: plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java (hunk 1 above)
Comment (dadoonet) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: Indeed. This makes sense.

Comment (dadoonet) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: Actually the filter is applied with a wildcard:

```java
        settingsFilter.addFilter("cloud.azure.storage.*");
```

I think that's why I did not explicitly filter it.  Does it make sense?

**Comment by dadoonet** [comment]
@imotov I added a comment. Not sure you saw it so I'm pinging you. :D

File: plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java (hunk 1 above)
Comment (imotov) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: Maybe it makes sense to replace it with `settingsFilter.addFilter(Storage. PREFIX + "*");` then? And what about `repositories.azure.account`? Any reason we want to hide `cloud.azure.storage.account` but not `repositories.azure.account`?

Comment (dadoonet) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: ++ I thought about using this static field as well.

We definitely need to hide also the repository one. Will update. Thanks.

Comment (dadoonet) [inline plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: I was thinking about it more and actually I think we should not hide all settings but only the ones ending with `account` or `key`.
So I'll try with that:

```java
settingsFilter.addFilter(Storage.PREFIX + "*.account");
settingsFilter.addFilter(Storage.PREFIX + "*.key");
settingsFilter.addFilter(Storage.ACCOUNT);
```

**Comment by dadoonet** [comment]
@imotov I added another commit. Could you review please? Just to make sure it's correct before I push it.

**Comment by imotov** [comment]
LGTM

**Pull request merged**